sdk/webpubsub/azure-messaging-webpubsub/src/samples/java/com/azure/messaging/webpubsub/BroadcastingWithFilterSample.java

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.webpubsub;
+
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.Configuration;
+
+public class BroadcastingWithFilterSample {
+    private static final String CONNECTION_STRING = Configuration.getGlobalConfiguration().get("WEB_PUB_SUB_CS");
+
+    public static void main(String[] args) {
+        // build a sync client
+        WebPubSubServiceClient chatHub = new WebPubSubServiceClientBuilder()
+            .connectionString(CONNECTION_STRING)
+            .hub("chat")
+            .buildClient();
+
+        // send a text message to the entire hub with a filter on userId
+        chatHub.sendToAllWithResponse(
+            BinaryData.fromString("Hello World - Broadcast test!"),
+            new RequestOptions().setHeader("Content-Type", "text/plain")
+                .addQueryParam("filter", "userId ne 'user1'"));
+
+        // send a text message to the entire hub with another filter on group
+        chatHub.sendToAllWithResponse(
+            BinaryData.fromString("Hello World - Broadcast test!"),
+            new RequestOptions().setHeader("Content-Type", "text/plain")
+                .addQueryParam("filter", "'GroupA' in groups and not('GroupB' in groups)"));
```

Comment:
Do we have API taking `WebPubSubContentType contentType`? That should be preferable API to call, than `setHeader`

Reply:
Yes we have the API, I will update the samples.

Reply:
One problem is that, the API with `WebPubSubContentType` and `RequestOptions` needs to take contentLength as well. (refer to below code snippet) I found it may be a bit hard for user to provide the contentLength. Also we need to add the query params using `new RequestOptions().addQueryParam()` to add filters, so we still need to create `RequestOptions`. My opinion is keeping the current code is easier for user since they don't need to provide contentLength. What do you think? 
```
public Response<Void> sendToAllWithResponse(
            BinaryData message,
            WebPubSubContentType contentType,
            long contentLength,
            RequestOptions requestOptions)
```

Reply:
contentLength is just `BinaryData.getLength` (for most case, except streaming)?

The problem with the current API is that no doc at all say user need to specify `content-type` header, and which value is allowed?